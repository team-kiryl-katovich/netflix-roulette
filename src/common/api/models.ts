export type Result<T> = {
  data?: T;
  error?: Error;
};

export type Offset = {
  offset?: number;
  limit?: number;
};

export type OffsetResult<T> = Result<T> &
  Offset & {
    total?: number;
  };

export type Movie = {
  id: number;
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
};

export type SortBy = keyof Movie;
export type SortOrder = 'desc' | 'asc';
export type SearchBy = Extract<keyof Movie, 'title' | 'genres'>;

export type MovieFilter = {
  sortBy?: SortBy;
  sortOrder?: SortOrder;
  search?: string;
  searchBy?: SearchBy;
  filter?: string;
};
