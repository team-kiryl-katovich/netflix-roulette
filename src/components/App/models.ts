import { Store } from 'redux';

import { AppState } from '../../store/models';

export type AppProps = {
  store: Store<AppState>;
};
