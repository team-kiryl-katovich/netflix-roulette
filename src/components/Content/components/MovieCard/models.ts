export type MovieCardProps = {
  id: number;
  title: string;
  gengre: string;
  year: number;
  imageUrl: string;
};
