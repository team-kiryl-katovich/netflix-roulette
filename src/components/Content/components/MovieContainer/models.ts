import { WithMoviesProps } from 'src/enhancers/withMovies';

export type MovieContainerComponentProps = WithMoviesProps;
