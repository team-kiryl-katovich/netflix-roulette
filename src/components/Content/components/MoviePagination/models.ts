import { Movie, Offset } from '@common/api';

export type MoviePaginationProps = {
  moviesCount: number;
  offset: Offset;
  pageSize: number;
  loading: boolean;
  children?: React.ReactNode;
  fetch: (offset: Offset) => void;
};

export type MoviePaginationState = {};
