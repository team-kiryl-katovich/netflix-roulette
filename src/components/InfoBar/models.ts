import { WithFilterProps } from '@enhancers/withFilter';
import { WithMoviesProps } from '@enhancers/withMovies';

export type InfoBarComponentProps = WithFilterProps & WithMoviesProps;
