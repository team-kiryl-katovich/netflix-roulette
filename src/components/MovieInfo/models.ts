import { WithMovieProps } from '@enhancers/withMovie';

export type MovieInfoComponentProps = WithMovieProps;
