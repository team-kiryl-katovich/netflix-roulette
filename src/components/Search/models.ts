import { WithMoviesHandlersProps } from '@enhancers/withMovies';
import { WithFilterProps } from '@enhancers/withFilter';

export type SearchComponentProps = WithFilterProps & WithMoviesHandlersProps;
