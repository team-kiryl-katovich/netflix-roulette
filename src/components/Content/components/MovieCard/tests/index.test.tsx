import * as React from 'react';
import * as enzyme from 'enzyme';

import { MovieCardProps } from '../models';
import { MovieCard } from '..';

describe('[COMPONENTS] [CONTENT] [COMPONENTS] [MOVIE_CARD]', () => {
  test('should render component', () => {
    const fakeProps: MovieCardProps = {
      gengre: 'fakeGengre',
      id: 1,
      imageUrl: 'fakeImageUtr',
      title: 'fakeTitle',
      year: 2019,
    };

    const component = enzyme.shallow(<MovieCard {...fakeProps} />);
    expect(component).toMatchSnapshot();
  });
});
