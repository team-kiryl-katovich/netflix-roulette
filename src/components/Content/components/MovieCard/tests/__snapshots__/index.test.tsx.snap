// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] [CONTENT] [COMPONENTS] [MOVIE_CARD] should render component 1`] = `
<div
  className="movie-card"
>
  <Link
    to="/movie/1"
  >
    <Component
      ratio={140}
      src="fakeImageUtr"
    />
    <div
      className="movie-card__description"
    >
      <div
        className="movie-card-title"
      >
        <Component
          className="movie-card-title__title"
          text="fakeTitle"
        />
        <Component
          className="movie-card-title__gengre"
          text="fakeGengre"
        />
      </div>
      <div
        className="movie-card-year"
      >
        <Component
          className="movie-card-year__label"
          text="2019"
        />
      </div>
    </div>
  </Link>
</div>
`;
